app.py

```diff
@@ -102,357 +125,270 @@ def toggle_left_column(left_column_collapse: int, class_name: str) -> str:
+    Output("dwave-tab", "disabled", allow_duplicate=True),
+    Output("mip-tab", "disabled", allow_duplicate=True),
+    Output("dwave-tab", "className", allow_duplicate=True),
+    Output("mip-tab", "className", allow_duplicate=True),
+    Output("run-button", "className", allow_duplicate=True),
+    Output("cancel-button", "className", allow_duplicate=True),
+    Output("running-dwave", "data", allow_duplicate=True),
+    Output("running-classical", "data", allow_duplicate=True),
+    Output("tabs", "value"),
+    [
+        Input("run-button", "n_clicks"),
+        Input("cancel-button", "n_clicks"),
+        State("solver-select", "value"),
+    ],
 )
-def load_initial_figures(
-    n_clicks: int,
-) -> tuple[go.Figure, go.Figure, go.Figure, go.Figure, dict, dict]:
-    """Loads the initial figures for the Gantt charts.
+def update_tab_loading_state(
+    run_click: int, cancel_click: int, solvers: list[str]
+) -> tuple[str, str, bool, bool, str, str, str, str, bool, bool, str]:
+    """Updates the tab loading state after the run button
+    or cancel button has been clicked.
 
     Args:
-        n_clicks (int): The number of times the run button has been
-        clicked.
+        run_click (int): The number of times the run button has been clicked.
+        cancel_click (int): The number of times the cancel button has been clicked.
+        solvers (list[str]): The list of selected solvers.
 
     Returns:
-        go.Figure: Gantt chart for the D-Wave hybrid solver
-        go.Figure: Gantt chart for the Classical solver
-        go.Figure: Results table for the D-Wave hybrid solver
-        go.Figure: Results table for the Classical solver
-        dict: Style for the D-Wave summary table
-        dict: Style for the Classical summary table
+        str: The label for the D-Wave tab.
+        str: The label for the Classical tab.
+        bool: True if D-Wave tab should be disabled, False otherwise.
+        bool: True if Classical tab should be disabled, False otherwise.
+        str: Class name for the D-Wave tab.
+        str: Class name for the Classical tab.
+        str: Run button class.
+        str: Cancel button class.
+        bool: Whether Hybrid is running.
+        bool: Whether MIP is running.
+        str: The value of the tab that should be active.
     """
-    if n_clicks == 0:
-        empty_figure = get_empty_figure("Run optimization to see results")
-        empty_table = generate_output_table(0, 0, 0)
+
+    if ctx.triggered_id == "run-button" and run_click > 0:
+        run_hybrid = SamplerType.HYBRID.value in solvers
+        run_mip = SamplerType.MIP.value in solvers
+
         return (
-            empty_figure,
-            empty_figure,
-            empty_table,
-            empty_table,
-            {"visibility": "hidden"},
-            {"visibility": "hidden"},
+            "Loading..." if run_hybrid else "D-Wave Results",
+            "Loading..." if run_mip else "Classical Results",
```

Comment:
This can return `dash.no_update` if not loading, since it only enters this callback on button click and the cancel click returns the correct labels already, no?
```suggestion
            "Loading..." if run_hybrid else dash.no_update,
            "Loading..." if run_mip else dash.no_update,
```